api/proto/teleport/legacy/types/events/events.proto

```diff
@@ -3693,6 +3695,56 @@ message ElasticsearchRequest {
   string Query = 12 [(gogoproto.jsontag) = "query"];
 }
 
+// AppSessionDynamoDBRequest is emitted when a user executes a DynamoDB request via app access.
+message AppSessionDynamoDBRequest {
+  // Metadata is a common event metadata.
+  Metadata Metadata = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+  // User is a common user event metadata.
+  UserMetadata User = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+  // App is a common application resource metadata.
+  AppMetadata App = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+  // AWS contains extra AWS metadata of the request.
+  AWSRequestMetadata AWS = 4 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+
+  // SessionChunkID is the ID of the app session chunk that this request belongs to.
+  // This is more appropriate to include than the app session id, since it is the chunk
+  // id that is needed to play back the session chunk with tsh. The session chunk event
+  // already includes the app session id.
+  string SessionChunkID = 5 [(gogoproto.jsontag) = "session_chunk_id"];
+  // StatusCode the HTTP response code for the request.
+  uint32 StatusCode = 6 [(gogoproto.jsontag) = "status_code"];
+  // Path is relative path in the URL.
+  string Path = 7 [(gogoproto.jsontag) = "path"];
+  // RawQuery are the encoded query values.
+  string RawQuery = 8 [(gogoproto.jsontag) = "raw_query"];
+  // Method is the request HTTP method, like GET/POST/DELETE/etc.
+  string Method = 9 [(gogoproto.jsontag) = "method"];
+  // Target is the API target in the X-Amz-Target header.
+  string Target = 10 [(gogoproto.jsontag) = "target"];
+  // Body is the HTTP request json body.
+  // The Struct type is a wrapper around protobuf/types.Struct and is used to marshal the JSON body correctly.
+  google.protobuf.Struct Body = 11 [
+    (gogoproto.jsontag) = "body,omitempty",
+    (gogoproto.casttype) = "Struct"
```

Comment:
Custom proto serialization via `casttype` can fail, it might be good idea to test this round trips etc.

Reply:
This is the same way we serialize IdentityAttributes in UserLogin message, which is where I got the idea to do this. I have manually verified that it serializes/deserializes properly.